### PR TITLE
Generated sparkle feedurl on runtime

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/16076

and use `--update-feed-url=custom_feed_url` switch to use custom feed url.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Intall universal build on intel macOS
2. Build universal build with this PR and upload to test update server
3. Update to version of step 2
4. Build newer non universal build and upload to test update server
5. Update and check whether update is done with image of step 4

### CODE REVIEW

File: build/mac/tweak_info_plist.py [hunk 1]
```diff
@@ -110,9 +108,6 @@ def Main(argv):
 
   plist['CrProductDirName'] = options.brave_product_dir_name
 
-  if options.brave_feed_url:
```
Comment (mihaiplesa) [inline build/mac/tweak_info_plist.py:113]: so we're going to lose the option to override? QA team might need a way to test updates on internal channels

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: Ah, I see. I'll revert this change and will use this if provided over generated url.
I'll f/u tomorrow.

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: QA team edits Info.plist to use intenal url?

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: Curious which one is more convenient among editing Info.plist or passing url as command lind args.

Comment (mihaiplesa) [reply build/mac/tweak_info_plist.py:113]: cc @brave/legacy_qa

Comment (mihaiplesa) [reply build/mac/tweak_info_plist.py:113]: Command line arg better since it doesn't change the hash or the contents of the app dir and it's more visible.

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: Added `--update-feed-url` switch.

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: @mihaiplesa,  should we get security review for this change?

Comment (mihaiplesa) [reply build/mac/tweak_info_plist.py:113]: I think we're fine.

Comment (simonhong) [reply build/mac/tweak_info_plist.py:113]: Ok!

**Review by mihaiplesa — APPROVED** [review]

**Pull request merged**